src/plugin-client.ts

```diff
@@ -61,33 +82,31 @@ class PluginClient extends EventEmitter {
       port,
       this.onMsg.bind(this),
       `IpcSocket(${this.pluginId}):`,
-      {verbose: this.verbose}
+      { verbose: this.verbose }
     );
-    this.ipcSocket.connectPromise.then((ws) => {
+    this.ipcSocket?.getConnectPromise()?.then((ws) => {
       this.ws = ws;
 
       // Register ourselves with the server
       this.verbose &&
         console.log(this.logPrefix, 'Connected to server, registering...');
 
-      this.sendNotification(MessageType.PLUGIN_REGISTER_REQUEST);
+      this.sendNotification((<any>MessageType).PLUGIN_REGISTER_REQUEST);
     });
 
-    return this.deferredReply.promise;
+    return this.deferredReply.getPromise();
   }
 
-  sendNotification(messageType, data = {}) {
+  sendNotification(messageType: string, data: any = {}) {
```

Comment:
Unless I'm mistaken, `messageType` should be a `number`.

Reply:
Done.